projects/packages/forms/src/blocks/contact-form/components/shared/plugin-integration-panel/index.js

```diff
@@ -0,0 +1,173 @@
+import { useAnalytics } from '@automattic/jetpack-shared-extension-utils';
+import { Button, Icon, Spinner, PanelBody } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { installAndActivatePlugin, activatePlugin } from '../../../util/plugin-management';
+import './styles.css';
+
+const PluginIntegrationPanel = ( {
+	pluginSlug,
+	pluginPath,
+	installText,
+	activateText,
+	description,
+	tracksEventName,
+	children,
+	title,
+	initialOpen = false,
+} ) => {
+	const [ isInstalling, setIsInstalling ] = useState( false );
+	const [ isActivating, setIsActivating ] = useState( false );
+	const { invalidateResolution } = useDispatch( coreStore );
+	const { tracks } = useAnalytics();
+
+	const { pluginStatus, isLoading } = useSelect(
+		select => {
+			const installedPlugins = select( coreStore ).getPlugins();
```

Comment:
I'm using WP core data store to determine plugin status. In the files for the Jetpack CRM and Creative Mail plugins, we appear to use a Jetpack endpoint. Example: 

https://github.com/Automattic/jetpack/blob/6deb2860206736cfdf2279be79b132693277cbe0/projects/packages/forms/src/blocks/contact-form/components/jetpack-crm-integration/jetpack-crm-integration-settings.js#L8

What's the best approach here?

Reply:
Here's something similar that might be reusable:

https://github.com/Automattic/jetpack/blob/6deb2860206736cfdf2279be79b132693277cbe0/projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.jsx#L367-L391

Reply:
If it works, I think I'll use the existing implementation.